x-pack/plugins/monitoring/server/plugin.ts

```diff
@@ -255,14 +255,14 @@ export class MonitoringPlugin
     }
   }
 
-  stop() {
+  async stop() {
```

Comment:
async setup/start/stop lifecycle methods are deprecated and should be avoided when possible. WDYT if we abort the request instead of waiting for the promise to complete?